splash/qtrender_image.py

```diff
@@ -457,13 +465,13 @@ def crop(self, rect):
         self.img = self.img.crop((left, top, right, bottom))
 
     def to_png(self, complevel=defaults.PNG_COMPRESSION_LEVEL):
-        buf = StringIO()
+        buf = BytesIO()
```

Comment:
It could be the case difference is non-existing in practice, but could you please try it with a larger image, something like 1024x40000px? Such images are common when render_all=1 is used.